### PR TITLE
fix grpc set metadata multi-value

### PR DESCRIPTION
**Pull request by wgy035**
Resolves https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/11237
`removeAll` key before `put` in grpc `MetadataSetter`

### CODE REVIEW

**Comment by steverao** [comment]
@wgy035 Please sign the CLA.

**Comment by trask** [comment]
@wgy035 can you add a test for this? (check out #2727 for possible inspiration)

File: instrumentation/grpc-1.6/library/src/main/java/io/opentelemetry/instrumentation/grpc/v1_6/MetadataSetter.java [hunk 1]
```diff
@@ -13,6 +13,7 @@ enum MetadataSetter implements TextMapSetter<Metadata> {
 
   @Override
   public void set(Metadata carrier, String key, String value) {
+    carrier.removeAll(Metadata.Key.of(key, Metadata.ASCII_STRING_MARSHALLER));
```
Comment (laurit) [inline instrumentation/grpc-1.6/library/src/main/java/io/opentelemetry/instrumentation/grpc/v1_6/MetadataSetter.java:16]: you could create the key only once and use it for both calls

**Comment by wgy035** [comment]
Got it. And now I'm trying to add a test.

File: instrumentation/grpc-1.6/library/src/test/java/io/opentelemetry/instrumentation/grpc/v1_6/propagator/GrpcPropagator.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.propagation.TextMapGetter;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import io.opentelemetry.context.propagation.TextMapSetter;
+import java.util.Collections;
+import java.util.List;
+
+public class GrpcPropagator implements TextMapPropagator {
+
+  public static final String FIELD = "X-grpc-field";
```
Comment (steverao) [inline instrumentation/grpc-1.6/library/src/test/java/io/opentelemetry/instrumentation/grpc/v1_6/propagator/GrpcPropagator.java:17]:
```suggestion
  private static final String FIELD = "X-grpc-field";
```

File: instrumentation/grpc-1.6/library/src/test/java/io/opentelemetry/instrumentation/grpc/v1_6/propagator/GrpcPropagator.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  private static final String METADATA = "value";
+
+  @Override
+  public List<String> fields() {
+    return Collections.singletonList(FIELD);
```
Comment (steverao) [inline instrumentation/grpc-1.6/library/src/test/java/io/opentelemetry/instrumentation/grpc/v1_6/propagator/GrpcPropagator.java:22]: Suggest to use static import here, for more style guidance you can refer to 
https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/contributing/style-guideline.md#static-imports

File: instrumentation/grpc-1.6/library/src/test/java/io/opentelemetry/instrumentation/grpc/v1_6/MetadataSetterTest.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import io.grpc.Metadata;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.grpc.v1_6.propagator.GrpcPropagator;
+import org.junit.jupiter.api.Test;
+
+class MetadataSetterTest {
+
+  @Test
+  void checkThatIterableSizeEqualsOne() {
+    GrpcPropagator tested = new GrpcPropagator();
```
Comment (steverao) [inline instrumentation/grpc-1.6/library/src/test/java/io/opentelemetry/instrumentation/grpc/v1_6/MetadataSetterTest.java:19]: I would define a more meaningful name here, if it's hard to do here, is the `propagator` a better choice than now?

File: instrumentation/grpc-1.6/library/src/test/java/io/opentelemetry/instrumentation/grpc/v1_6/propagator/GrpcPropagator.java (hunk 2 above)
Comment (wgy035) [reply instrumentation/grpc-1.6/library/src/test/java/io/opentelemetry/instrumentation/grpc/v1_6/propagator/GrpcPropagator.java]: done

File: instrumentation/grpc-1.6/library/src/test/java/io/opentelemetry/instrumentation/grpc/v1_6/propagator/GrpcPropagator.java (hunk 3 above)
Comment (wgy035) [reply instrumentation/grpc-1.6/library/src/test/java/io/opentelemetry/instrumentation/grpc/v1_6/propagator/GrpcPropagator.java]: helpful suggestions

File: instrumentation/grpc-1.6/library/src/test/java/io/opentelemetry/instrumentation/grpc/v1_6/MetadataSetterTest.java (hunk 4 above)
Comment (wgy035) [reply instrumentation/grpc-1.6/library/src/test/java/io/opentelemetry/instrumentation/grpc/v1_6/MetadataSetterTest.java]: I've already change the name

**Comment by laurit** [comment]
@wgy035 I have crated a PR for your PR https://github.com/wgy035/opentelemetry-java-instrumentation/pull/1 that changes the test so that it uses a standard propagator and verifies that injecting the tracing context twice results in the last inject overwriting the first one.

**Review by laurit — APPROVED** [review]

**Pull request merged**